pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java

```diff
@@ -329,27 +329,30 @@ public static <R, X extends Exception> R withTxn(HandleCallback<R, X> callback)
      * values so that we can rotate the password without rebooting the app.
      */
     private static Handle openJdbiWithAuthRetry(DB db) {
+        Exception lastThrownException = null;
         for (int tryCount = 0; tryCount < PASSWORD_ROTATION_MAX_RETRIES; tryCount++) {
             try {
                 return getInstance(db).jdbi.open();
             } catch (ConnectionException e) {
+                lastThrownException = e;
                 if (isAuthException(e)) {
-                    log.info("Database pool credentials have been rejected; pausing and reloading creds from config file.", e);
+                    log.info("Database pool credentials have been rejected; pausing and reloading credentials from config file.", e);
                     try {
                         TimeUnit.SECONDS.sleep(PASSWORD_ROTATION_SLEEP);
                     } catch (InterruptedException interrupted) {
-                        log.error("Sleep before dbpool credential reload has been interrupted", interrupted);
+                        log.error("Sleep before dbpool credential reload has been interrupted.", interrupted);
                     }
                     reloadDbPoolConfiguration(false);
                 } else {
-                    throw new DDPException(COULD_NOT_GET_CONNECTION, e);
+                    log.info("Open jdbi connection retried {} time.", tryCount + 1);
```

Comment:
Should I add `TimeUnit.SECONDS.sleep(PASSWORD_ROTATION_SLEEP)` here before next retry?